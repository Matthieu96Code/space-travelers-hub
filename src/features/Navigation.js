import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../assets/planet.png';

const Navigation = () => {
  const title = "Space Travelers' Hub";

  return (
    <nav className="navbar navbar-expand-lg bg-body-tertiary">
      <div className="container py-2">
        <a className="navbar-brand d-flex align-items-center gap-3 fw-semibold" href="/">
          <img src={logo} width="50px" alt={title} />
          <span className="d-none d-sm-none d-md-block d-lg-block">{title}</span>
        </a>
        <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse justify-content-end" id="navbarSupportedContent">
          <ul className="navbar-nav mb-2 mb-lg-0 gap-2 gap-sm-2 gap-md-2 gap-lg-4">
            <li className="nav-link">
              <NavLink to="rockets"> Rockets</NavLink>
            </li>

            <li className="nav-link">
              <NavLink to="missions"> Missions</NavLink>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
};

export default Navigation;
