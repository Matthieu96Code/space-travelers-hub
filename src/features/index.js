export { default as Navigation } from './Navigation';
export { default as MissionsTable } from './missions/MissionsTable';
export { default as RocketsList } from './rockets/RocketsList';
