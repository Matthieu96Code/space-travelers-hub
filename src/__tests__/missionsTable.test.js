/* eslint-disable no-unused-vars */
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { MissionsTable } from '../features';

const mockStore = configureMockStore();
const store = mockStore({
  mission: {
    name: 'mission',
    missions: [
      {
        mission_name: 'Thaicom',
        manufacturers: ['Orbital'],
        payload_ids: ['1', '2'],
        mission_id: '9D1B7E0',
        wikipedia: 'https://en.wikipedia.org/wiki/Thaicom',
        website: 'http://www.thaicom.net/en/satellites/overview',
        twitter: 'https://twitter.com/thaicomplc',
        description: 'Thaicom is the name of a series of communications satellites operated from Thailand',
        reserved: false,
      },
    ],
  },
});

describe('Missions Table', () => {
  it('component should render correctly', () => {
    render(
      <Provider store={store}>
        <MissionsTable />
      </Provider>,
    );

    const missionsTableElement = screen.getByRole('table');
    expect(missionsTableElement).toBeInTheDocument();
  });

  it('render list of missions', async () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MissionsTable />
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
