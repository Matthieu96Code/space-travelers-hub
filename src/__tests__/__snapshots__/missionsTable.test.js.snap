// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Table render list of missions 1`] = `
<div
  className="missions-table overflow-x-auto"
>
  <table
    className="table table-striped table-responsive table-bordered"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Mission
        </th>
        <th
          scope="col"
        >
          Description
        </th>
        <th
          scope="col"
        >
          Status
        </th>
        <th
          scope="col"
        />
      </tr>
    </thead>
    <tbody>
      <tr
        className="mission_info"
      >
        <th
          scope="row"
          style={
            Object {
              "minWidth": "8rem",
            }
          }
        >
          Thaicom
        </th>
        <td>
          Thaicom is the name of a series of communications satellites operated from Thailand
        </td>
        <td>
          <span
            className="badge text-bg-secondary fw-semibold"
          >
            NOT A MEMBER
          </span>
        </td>
        <td
          style={
            Object {
              "minWidth": "10rem",
              "textAlign": "center",
            }
          }
        >
          <button
            className="btn btn-outline-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            Join Mission
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
